File: Knjigoteka.Model/Entities/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Knjigoteka.Model.Entities
{
    public class Employee
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int UserId { get; set; }
        [Required]
        public int BranchId { get; set; }
        [Required]
        public DateTime EmploymentDate { get; set; } = DateTime.Now;
        [Required]
        public bool IsActive { get; set; }
        [ForeignKey(nameof(UserId))]
        public User User { get; set; } = null!;
        [ForeignKey(nameof(BranchId))]
        public Branch Branch { get; set; } = null!;
    }
}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/employee.dart
class Employee {
  final int id;
  final int userId;
  final String fullName;
  final int branchId;
  final String branchName;
  final String employmentDate;
  final bool isActive;

  Employee({
    required this.id,
    required this.userId,
    required this.fullName,
    required this.branchId,
    required this.branchName,
    required this.employmentDate,
    required this.isActive,
  });

  factory Employee.fromJson(Map<String, dynamic> json) => Employee(
    id: json['id'],
    userId: json['userId'],
    fullName: json['fullName'],
    branchId: json['branchId'],
    branchName: json['branchName'],
    employmentDate: json['employmentDate'],
    isActive: json['isActive'] ?? true,
  );
}
